src/npc.h

```diff
@@ -221,6 +227,7 @@ class Npc final : public Creature
 		std::map<std::string, std::string> m_parameters;
 
 		std::set<Player*> shopPlayerSet;
+		std::set<Player*> spectators;
```

Comment:
Fair enough, it would actually be hard to do that properly otherwise. However, the complexity of a find in std::set is O(logn). In this case an unordered_set __might__ be a better choice.